src/routes.js

```diff
@@ -114,41 +128,49 @@ export const routes = [
   {
     path: '/:chapterId(\\d+)/:range/:translations',
     component: loadable(() => import('./containers/Surah')),
-    loadData: [chaptersConnect, chapterInfoConnect, versesConnect]
-    // import('./components/GlobalNav/Surah')
-    // onEnter={checkValidChapterOrVerse}
+    loadData: [chaptersConnect, chapterInfoConnect, versesConnect],
+    navbar: loadable(() =>
+      import(/* webpackChunkName: "globalnav-surah" */ './components/GlobalNav/Surah')
+    ),
+    onEnter: checkValidChapterOrVerse
   },
   {
     path: '/:chapterId(\\d+)/:range?.pdf',
     component: loadable(() =>
       import(/* webpackChunkName: "pdf" */ './containers/Pdf')
     ),
-    loadData: [chaptersConnect, versesConnect]
-    // import(
-    //   /* webpackChunkName: "pdf-footer" */ './components/Footer/PdfFooter'
-    // )
-    // onEnter={checkValidChapterOrVerse}
+    loadData: [chaptersConnect, versesConnect],
+    footer: loadable(() =>
+      import(/* webpackChunkName: "pdf-footer" */ './components/Footer/PdfFooter')
+    ),
+    onEnter: checkValidChapterOrVerse
   },
   {
     path: '/:chapterId(\\d+)/:range?',
     component: loadable(() =>
       import(/* webpackChunkName: "surah" */ './containers/Surah')
     ),
-    loadData: [chaptersConnect, chapterInfoConnect, versesConnect]
-    // import(
-    //   /* webpackChunkName: "globalnav-surah" */ './components/GlobalNav/Surah'
-    // )
-    // onEnter={checkValidChapterOrVerse}
+    loadData: [chaptersConnect, chapterInfoConnect, versesConnect],
+    navbar: loadable(() =>
+      import(/* webpackChunkName: "globalnav-surah" */ './components/GlobalNav/Surah')
+    ),
+    onEnter: checkValidChapterOrVerse
   }
 ];
 
 const Routes = ({ store }) =>
   <Switch>
-    {routes.map(({ component: Component, loadData, ...route }) =>
+    {routes.map(({ component: Component, loadData, setContext, ...route }) =>
       <Route
         key={route.path}
         {...route}
-        render={(routeProps) => {
+        render={({ staticContext, ...routeProps }) => {
+          if (staticContext) {
+            const contextFunction = setContext || defaultSetContext;
+
+            Object.assign(staticContext, contextFunction(staticContext));
+          }
+
           if (__CLIENT__) {
             routePromises({
```

Comment:
can we catch this promise and return an error page?

Reply:
Good catch. I will add it inshallah